modules/system-source/system-source.c

```diff
@@ -186,33 +186,29 @@ _is_fd_pollable(gint fd)
 static gboolean
 _detect_linux_dev_kmsg(void)
 {
-  gint fd;
+  gint fd = open("/dev/kmsg", O_RDONLY);
 
-  if ((fd = open("/dev/kmsg", O_RDONLY)) != -1)
-    {
-      if ((lseek (fd, 0, SEEK_END) != -1) && _is_fd_pollable(fd))
-        {
-          return TRUE;
-        }
-      close (fd);
-    }
-  return FALSE;
+  if (fd == -1)
+    return FALSE;
+
+  gboolean seekable_and_pollable = (lseek(fd, 0, SEEK_END) != -1) && _is_fd_pollable(fd);
```

Comment:
why not having two independent gbooleans?
```
gboolean seekable =...
gboolean pollable = ...
close(fd);

return seekable && pollable;
```

Reply:
Good idea, thanks.